lib/archethic/p2p/message/start_mining.ex

```diff
@@ -96,12 +100,23 @@ defmodule Archethic.P2P.Message.StartMining do
         welcome_node_public_key: welcome_node_public_key,
         validation_node_public_keys: validation_node_public_keys,
         network_chains_view_hash: network_chains_view_hash,
-        p2p_view_hash: p2p_view_hash
+        p2p_view_hash: p2p_view_hash,
+        contract_context: contract_context
       }) do
+    serialized_contract_context =
+      case contract_context do
```

Comment:
This condition is often checked, I think it could be easier to accept that a Contract.Context can be nil, so the functions `Contract.Context.serialize` or `deserialize` could return a specific value in case of nil context
```elixir
def serialize(nil), do: <<0::8>>
def serialize(context) do <<1::8, ...>> end

def deserialize(<<0::8, rest::binary>>), do: {nil, rest}
def deserialize(<<1::8, rest::binary>>) do ... end
```

Reply:
What's the point of serialize an nil structure ?
It should be handled upfront.

Reply:
It's in case the context is not the last part of a struct that contains the context (StartMining/NewTransaction)

Reply:
Just need to put a bit in the newTx and startling, like in the transaction struct with validation stamp

Reply:
I reverted to what  it used to be